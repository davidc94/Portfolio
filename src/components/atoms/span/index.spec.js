import React from 'react';
import renderer from 'react-test-renderer';
import StyledSpan from './index';
import Colors from '../../../utils/colors';

describe('StyledSpan component', () => {
  it('render default StyledSpan as expected', () => {
    const render = renderer.create(
      <StyledSpan>I have nothing to offer</StyledSpan>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('render StyledSpan with a different background color', () => {
    const render = renderer.create(
      <StyledSpan backgroundColor={Colors.blue}>nice background color</StyledSpan>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
  
  it('render StyledSpan with a different color', () => {
    const render = renderer.create(
      <StyledSpan color={Colors.lightBlue}>cool blue</StyledSpan>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('render StyledSpan with a different font size', () => {
    const render = renderer.create(
      <StyledSpan size="large">EXTRA BIG</StyledSpan>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});