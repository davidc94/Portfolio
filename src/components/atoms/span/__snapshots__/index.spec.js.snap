// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledSpan component render StyledSpan with a different background color 1`] = `
<span
  className="sc-bdVaJa cvQqPW"
>
  nice background color
</span>
`;

exports[`StyledSpan component render StyledSpan with a different color 1`] = `
<span
  className="sc-bdVaJa jvdlUN"
  color="#A8DADC"
>
  cool blue
</span>
`;

exports[`StyledSpan component render StyledSpan with a different font size 1`] = `
<span
  className="sc-bdVaJa dzrLNz"
  size="large"
>
  EXTRA BIG
</span>
`;

exports[`StyledSpan component render default StyledSpan as expected 1`] = `
<span
  className="sc-bdVaJa iKEBZc"
>
  I have nothing to offer
</span>
`;
